plans/selftest.fmf

```diff
@@ -0,0 +1,22 @@
+summary: Run make all target from Makefile using packet.dev CI
+
+prepare:
+  - name: packages
+    how: install
+    package:
+    # ~Minimum dependecies for configure script and makefile selftests
+    - bpftool
+    - clang
+    - elfutils-libelf
+    - elfutils-libelf-devel
+    - libbpf
+    - libbpf-devel
+    - llvm
+    - libpcap
+    - libpcap-devel
+    - m4
+    - sqlite-devel
+    - zlib-devel
+    - git
+
+test: ./test.sh
```

Comment:
Looks like using `test` results in a test case, but using `execute` in a plan... So you really don't have any plans to run...

I would suggest looking at here:
https://packit.dev/docs/testing-farm/

Or ask TMT people to help you with the setup.

Reply:
All depends how you want to handle the dependencies. The expected way is to have them defined in the test instead of the plan. So, how about following:
Have `test.fmf` file which defines test medata:
```yaml
summary: Run make all target from Makefile
test: ./test.sh
require:
    - bpftool
    - clang
    - elfutils-libelf
    - elfutils-libelf-devel
    - libbpf
    - libbpf-devel
    - llvm
    - libpcap
    - libpcap-devel
    - m4
    - sqlite-devel
    - zlib-devel
    - git
```
and use simplest plan called e.g. `plan.fmf`:
```
discover:
  how: fmf
execute:
 how: tmt
```

Reply:
Another way, for example if you don't want to hassle with dedicated file for test metadata could be to have everything defined in the plan file (any name will do, Plan is defined by having 'execute' key):
```yaml
execute:
    how: tmt

prepare:
  - name: packages
    how: install
    package:
    # ~Minimum dependecies for configure script and makefile selftests
    - bpftool
    - clang
    - elfutils-libelf
    - elfutils-libelf-devel
    - libbpf
    - libbpf-devel
    - llvm
    - libpcap
    - libpcap-devel
    - m4
    - sqlite-devel
    - zlib-devel
    - git

discover:
   how: shell
   tests:
       - name: Run make all target from Makefile
         test: ./test.sh
```

Note that you can use 'require' in test definition so another approach which just a plan metadata file is:
```yaml
execute:
    how: tmt

discover:
   how: shell
   tests:
       - name: Run make all target from Makefile
         test: ./test.sh
         require:
         - bpftool
         - clang
         - elfutils-libelf
         - elfutils-libelf-devel
         - libbpf
         - libbpf-devel
         - llvm
         - libpcap
         - libpcap-devel
         - m4
         - sqlite-devel
         - zlib-devel
         - git
```